ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -59,41 +61,41 @@ internal class PaywallViewModelImpl(
     preview: Boolean = false,
 ) : ViewModel(), PaywallViewModel {
 
-    private val variableDataProvider = VariableDataProvider(applicationContext, preview)
+    val variableDataProvider = VariableDataProvider(applicationContext, preview)
 
     override val state: StateFlow<PaywallViewState>
         get() = _state.asStateFlow()
+
+    val mode: PaywallViewMode
+        get() = options.mode
+
     private val _state: MutableStateFlow<PaywallViewState>
     private val _lastLocaleList = MutableStateFlow(getCurrentLocaleList())
     private val _colorScheme = MutableStateFlow(colorScheme)
 
-    private val offering: Offering?
-        get() = options.offeringSelection.offering
-
     private val listener: PaywallViewListener?
         get() = options.listener
 
-    private val mode: PaywallViewMode
-        get() = options.mode
-
     init {
-        _state = MutableStateFlow(offering?.calculateState() ?: PaywallViewState.Loading)
-        if (offering == null) {
-            updateOffering()
+        val selectedOffering = options.offeringSelection.offering
+        _state = MutableStateFlow(PaywallViewState.Loading)
+
+        if (selectedOffering == null) {
+            updateState()
```

Comment:
I've simplified this a lot because now we need to fetch `Offerings` and `CustomerInfo` (like in iOS). So there's now only a single path: `updateState` does all the logic

Reply:
Hmm I'm actually wondering, if `selectedOffering` is not null, where do we set the `_state` to the final value?

Reply:
Hmmm good catch, fixed.